Bryce's Submission

bryce-mcmath opened the pull request:
Chrome extension made with Vue and AceEditor, spent the majority of the time (3hrs) learning about Chrome extensions and AceEditor, didn't have time to do anything cool, other than detect dark mode preference, edit styling and editor theme accordingly, and ofcourse - actually edit CSS. Oh and I added a drop down for more editor modes that I didn't get time to use. 

## Gif:
![Demo](https://github.com/bryce-mcmath/2020-04-04-Live-CSS-Editor/blob/master/mintbean__demo.gif?raw=true) 

## Pics:
![Not in use](https://github.com/bryce-mcmath/2020-04-04-Live-CSS-Editor/blob/master/unused.png?raw=true)

![In use](https://github.com/bryce-mcmath/2020-04-04-Live-CSS-Editor/blob/master/used.png?raw=true)

bryce-mcmath committed:
Initial project setup

What:

1) Vue
2) Ace Editor
3) vue-web-extension

Why:

1) Easy reactive state
2) Easy editor
3) Easy chrome extension

bryce-mcmath committed:
Get editor working, add dropdown for lang select

bryce-mcmath committed:
MVP

What:

1) Add permissions to manifest
2) Add watcher for editor content, insertCSS on current tab

bryce-mcmath committed:
Add dark mode detection and basic styles

What:

1) Add computed property to detect preferred theme
2) Add class binding
3) Add basic styles

bryce-mcmath committed:
Merge branch 'master' of https://github.com/bryce-mcmath/2020-04-04-Live-CSS-Editor

bryce-mcmath committed:
Add gif

monarchwadia commented:
No screenshots or videos for me? :-) Can't score aesthetics without that.

monarchwadia commented on package.json:
```diff
@@ -0,0 +1,54 @@
+  },
+  "dependencies": {
+    "axios": "^0.19.0",
+    "emmet": "git+https://github.com/cloud9ide/emmet-core.git#41973fcc70392864c7a469cf5dcd875b88b93d4a",
```
are you using emmet in your editor?
if not, you can take emmet out of dependencies and drop it in devdependencies.
although i see that this is from cloud9ide, so maybe you ARE using it as a plugin or something for Ace

bryce-mcmath replied:
I am intentionally including emmet as a plugin for Ace. I was aiming to add html and js to the dropdown of options, but didn't have enough time

monarchwadia commented on scripts/build-zip.js:
```diff
@@ -0,0 +1,53 @@
+    .catch(console.err); 
+};
+
+main();
```
I guess this file is to wrap the extension up into a zip, for deployment on a browser? double checking :-)

bryce-mcmath replied:
Yessir

monarchwadia commented on src/popup/App.vue:
```diff
@@ -0,0 +1,94 @@
+      let tab;
+      chrome.tabs.query({ currentWindow: true, active: true }, tabs => {
+        chrome.tabs.insertCSS(tabs[0].id, { code: val });
+      });
```
this is slightly difficult to read. can we have comments or something a bit more explanatory?

bryce-mcmath replied:
Added comments and removed superfluous scaffolding

monarchwadia commented:
Awesome work! Can we please see a better video?

bryce-mcmath committed:
Add pics

montrealist commented on src/background.js:
```diff
@@ -0,0 +1,4 @@
+global.browser = require('webextension-polyfill');
```
This file can probably be safely deleted.

montrealist commented on src/popup/App.vue:
```diff
@@ -0,0 +1,94 @@
+    };
+  },
+  computed: {
+    dark() {
```
Adapting to the user's colour scheme, that's a really nice touch!

leah-h replied:
don't see the link to the project...but this sounds great

bryce-mcmath replied:
> don't see the link to the project...but this sounds great

There are photos and a gif in the repo. If you want to try it out I can show you how, just let me know!

montrealist commented on src/popup/App.vue:
```diff
@@ -0,0 +1,94 @@
+    content: function(val) {
+      let tab;
+      chrome.tabs.query({ currentWindow: true, active: true }, tabs => {
+        chrome.tabs.insertCSS(tabs[0].id, { code: val });
```
Will this always only work on the first browser tab (`tabs[0]`)?

bryce-mcmath replied:
Yes, that was the intent. I figured somebody using this would only want to apply to it one tab at a time

bryce-mcmath replied:
Added a comment to make it clear

montrealist commented on src/store/actions.js:
```diff
@@ -0,0 +1,5 @@
+import * as types from './mutation-types';
```
`setFoo` - this file looks like it's not used?

bryce-mcmath replied:
That's just Vuex scaffolding I didn't use and didn't have time to cleanup :)

bryce-mcmath replied:
It's been cleaned up!

montrealist reviewed (approved):
Good work! A bunch of `foo`s everywhere look like they should be deleted - unless it's your variable, but then it's better to name it something more meaningful.

Michael22936 commented:
Really amazing job!

KieranDevlin commented on src/popup/popup.html:
```diff
@@ -0,0 +1,17 @@
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Title</title>
```
I would add a relevant title here

derekjj commented:
Great job doing the challenge task!

bryce-mcmath committed:
Cleanup

What:

1) Remove Vuex scaffolding
2) Add title to html as suggested by code review
3) Add a few comments, delete unused variables

bryce-mcmath commented:
Thanks everyone, that was fun!

bryce-mcmath commented:
> No screenshots or videos for me? :-) Can't score aesthetics without that.

There are pictures and a gif in the repo :)

bryce-mcmath commented:
Added pics and gif explicitly to first PR post

bryce-mcmath committed:
Update Readme.md

bryce-mcmath committed:
Fix gif

bryce-mcmath committed:
Merge branch 'master' of https://github.com/bryce-mcmath/2020-04-04-Live-CSS-Editor

bryce-mcmath committed:
Update Readme.md
